tardis/conftest.py

```diff
@@ -75,22 +80,37 @@ def pytest_unconfigure(config):
             "{1}".format(githash, report_content)
         )
 
-        try:
-            doku_conn = dokuwiki.DokuWiki(
-                url=integration_tests_config["dokuwiki"]["url"],
-                user=integration_tests_config["dokuwiki"]["username"],
-                password=integration_tests_config["dokuwiki"]["password"])
-        except gaierror, dokuwiki.DokuWikiError:
-            print "Dokuwiki connection not established, report upload failed!"
-        else:
-            # Upload report on dokuwiki. Temporary link due to prototyping purposes.
-            doku_conn.pages.append("reports:{0}".format(githash[:7]), report_content)
-            print "Uploaded report on Dokuwiki."
+        # These steps are already performed by `integration_tests_config` and
+        # `doku_conn` fixtures, yet they are to be written here again, because
+        # all the fixtures are teared down and no longer usable, when this
+        # method is being called by pytest.
+        integration_tests_configpath = config.getvalue("integration-tests")
+        if integration_tests_configpath is not None:
+            integration_tests_configpath = os.path.expandvars(
+                os.path.expanduser(integration_tests_configpath)
+            )
+            integration_tests_config = yaml.load(open(
+                    integration_tests_configpath))
+
+            try:
+                doku_conn = dokuwiki.DokuWiki(
+                    url=integration_tests_config["dokuwiki"]["url"],
+                    user=integration_tests_config["dokuwiki"]["username"],
+                    password=integration_tests_config["dokuwiki"]["password"])
+            except gaierror, dokuwiki.DokuWikiError:
+                print "Dokuwiki connection not established, report upload failed!"
+            else:
+                # Upload report on dokuwiki. Temporary link due to prototyping purposes.
+                doku_conn.pages.set("reports:{0}".format(githash[:7]), report_content)
+                print "Uploaded report on Dokuwiki."
 
     # Remove the local report file. Keeping the report saved on local filesystem
     # is not desired, hence deleted.
     os.unlink(config.option.htmlpath)
     print "Deleted temporary file containing html report."
+    # Remove tempdir by recursive deletion
+    os.system("rm -r {0}".format(config.option.tempdir))
```

Comment:
`tempdir` won't be empty so `os.removedirs`, `os.rmdir`, `os.unlink` won't work. Hence I used `os.system` for recursive deletion.

Although there is a built-in module named `shutils` which provides high level file interactions, and recursive deletion of non-empty directory is handled as `shutil.rmtree("/path/to/dir")`. I was hesitant to use it and it is required just for one purpose, and we have an alternative (`os.system`), which quite clearly shows what the code is doing here. What should I do here - let it be or change it ?